tests/test_tcp.py

```diff
@@ -1255,6 +1256,144 @@ def prog(sock):
         for client in clients:
             client.stop()
 
+    def test_create_server_ssl_over_ssl(self):
+        if self.implementation == 'asyncio':
+            raise unittest.SkipTest('asyncio does not support SSL over SSL')
+
+        CNT = 0           # number of clients that were successful
+        TOTAL_CNT = 25    # total number of clients that test will create
+        TIMEOUT = 10.0    # timeout for this test
+
+        A_DATA = b'A' * 1024 * 1024
+        B_DATA = b'B' * 1024 * 1024
+
+        sslctx_1 = self._create_server_ssl_context(self.ONLYCERT, self.ONLYKEY)
+        client_sslctx_1 = self._create_client_ssl_context()
+        sslctx_2 = self._create_server_ssl_context(self.ONLYCERT, self.ONLYKEY)
+        client_sslctx_2 = self._create_client_ssl_context()
+
+        clients = []
+
+        async def handle_client(reader, writer):
+            nonlocal CNT
+
+            # hack reader and writer to call start_tls()
+            transport = writer._transport
+            writer._transport = None
+            reader._transport = None
```

Comment:
Wow, you're brave! :)

Reply:
LOL was rushing to avoid manual protocol code - perhaps it's better be replaced by protocol so that the test can be robust enough even if asyncio changes its implementation in the future.